Style/responsive menu

MichMatrix opened the pull request.

MichMatrix committed:
style: add responsive menu

add menu icon linked to JS function for action of display navbar items when click on this icon and also add classes in the html file.

MichMatrix committed:
style: add js funtion for responsive menu

when click on the icon it is possible to display with a good style the navbar items adding two different css classes

MichMatrix committed:
style: create a responsive menu style

add style for when screens are smallers under 950px it will display a menu icon and if click on the icon, it is possible to see the menu items of the navbar

Mindful-Developer commented on js/scripts.js:
```diff
@@ -72,4 +72,23 @@ numItemsInput.addEventListener("focusout", () => {
+/*function for responsive navbar menu*/
+homeButton.addEventListener("click", () => responsiveMenu());
+function responsiveMenu() {
+  var mobileMenu = document.getElementById("top-nav");
```
var is not too great to use any more. const and let are what replaced it. Can you switch the var keywords to const keywords?

MichMatrix replied:
Good to remember. Done.

Mindful-Developer commented on index.html:
```diff
@@ -17,12 +17,18 @@
+                        <li class="logo"><a href="index.html">Chuckster</a></li>
+                        <li class="nav-links" id="breadcrumb"></li>
+                    </ul>
+                    <a href="javascript:void(0);" class="mobile-menu" onclick="responsiveMenu()">
```
We've been using "#" as the href to not redirect in all the rest of the anchor tags.

Also, we've been using eventlisteners in the JS instead on onclick html functions to keep the JS separate from the HTML

MichMatrix replied:
Done

Mindful-Developer commented on css/styles.css:
```diff
@@ -479,52 +482,80 @@ footer a:hover {
         width: 234px;
         height: 45px;
     }
 
```
Is there any way we can add a transition effect to drop the menu down slowly? Maybe the same cubic bezier that I gave to the custom select?

MichMatrix replied:
Good to remember it, I love transition effect, because seems more interactive to the user. Done.

Mindful-Developer replied:
The expand looks amazing!! Can we also get a collapsing effect too?

Mindful-Developer reviewed (changes requested):
Some suggestions and fixes to keep things consistent. Mobile menu looks amazing though!!

MichMatrix committed:
fix: remove unused classes and properties

MichMatrix committed:
fix: remove addEventListener and unused class

remove addEventListener from responsive navbar section and remove the implement of unused class .responsive-nav from the responsiveMenu function. In addition, modify local variable to const instead of global scope var

MichMatrix committed:
fix: modify href of the menu

MichMatrix committed:
style: add animated properties to responsive menu

style: add transition and scale transform when click on menu icon

MichMatrix committed:
style: add transition to search input on menu

MichMatrix committed:
style: add more space when click on the menu

MichMatrix committed:
style: modify home mobile margin due new style menu

MichMatrix committed:
style: add collapse when click on menu items

MichMatrix committed:
style: append transition effect when close menu

Mindful-Developer reviewed (approved):
You are a rockstar!! Soo awesome!
